.github/workflows/test.yml

```diff
@@ -0,0 +1,121 @@
+name: TestCI
+
+on: [push]
+
+jobs:
+  quality:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v1
+      - name: Use Node.js 12
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12
+
+      - uses: actions/cache@v1
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-quality-npm-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-quality-npm-
+
+      - uses: actions/cache@v1
+        with:
+          path: ~/.cache
+          key: ${{ runner.os }}-quality-cache-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-quality-cache-
+
+      - name: Install Packages
+        run: |
+          npm ci
+          cd e2e
+          npm ci
+
+      - name: Code Quality
+        run: |
+          npm run format
+          bash scripts/ci-test-no-changes.sh 'you probably committed unformatted code'
+          node reports/tslint
+
+      - name: Compile
+        run: |
+          npm run ng -- build --aot
+          npx tsc -p tsconfig.spec.json
+          cd e2e
+          npx tsc -p cypress/tsconfig.json
+          cd ..
+          cd schematics
+          npm run build
+          bash ../scripts/ci-test-no-changes.sh 'you probably committed unformatted code'
```

Comment:
are subshells supported here? something like:
```
(cd schematics && npm run build)
bash scripts/ci-test-no-changes.sh 'you probably committed unformatted code'
```

would be more readable if the main execution flow does not change dirs as often